pilosa.go

```diff
@@ -171,3 +168,70 @@ func ContainsSubstring(a string, list []string) bool {
 	}
 	return false
 }
+
+// NormalizeAddress converts addr into a valid "IP4:port" string.
+func NormalizeAddress(addr string) (string, error) {
+	host, port, err := hostPortWithDefaults(addr)
+	if err != nil {
+		return "", err
+	}
+	return net.JoinHostPort(HostToIP(host), port), nil
+}
+
+// HostToIP converts host to an IP4 address based on net.LookupIP().
+func HostToIP(host string) string {
```

Comment:
@raskle `HostToIP` is covered with the tests on `NormalizeAddress`, but I added test coverage for `AddressWithDefaults`.